python/ray/data/block.py

```diff
@@ -114,6 +119,9 @@ def __init__(self):
         self.wall_time_s: Optional[float] = None
         self.cpu_time_s: Optional[float] = None
         self.node_id = ray.runtime_context.get_runtime_context().node_id.hex()
+        # Max memory usage. May be an overestimate since we do not
+        # differentiate from previous tasks on the same worker.
+        self.max_rss_bytes: int = 0
```

Comment:
Hmm could the peak usage actually be higher during the middle of the task?

Reply:
Maybe we should also record this at the start, so we know if the task increased it.